ballista/rust/core/src/execution_plans/shuffle_writer.rs

```diff
@@ -489,8 +491,27 @@ mod tests {
     use datafusion::physical_plan::expressions::Column;
     use datafusion::physical_plan::limit::GlobalLimitExec;
     use datafusion::physical_plan::memory::MemoryExec;
+    use std::borrow::Borrow;
     use tempfile::TempDir;
 
+    pub trait StructArrayExt {
+        fn column_names(&self) -> Vec<&str>;
+        fn column_by_name(&self, column_name: &str) -> Option<&ArrayRef>;
+    }
+
+    impl StructArrayExt for StructArray {
+        fn column_names(&self) -> Vec<&str> {
+            self.fields().iter().map(|f| f.name.as_str()).collect()
+        }
+
+        fn column_by_name(&self, column_name: &str) -> Option<&ArrayRef> {
+            self.fields()
+                .iter()
+                .position(|c| c.name() == &column_name)
+                .map(|pos| self.values()[pos].borrow())
+        }
+    }
+
```

Comment:
As suggested by Jorge in https://github.com/jorgecarleitao/arrow2/pull/416#pullrequestreview-757660064